clippy_utils/src/lib.rs

```diff
@@ -2777,12 +2777,13 @@ pub fn span_extract_comment(sm: &SourceMap, span: Span) -> String {
 /// Returns all the comments a given span contains.
 ///
 /// Comments are returned wrapped with their relevant delimiters.
+#[allow(clippy::unnecessary_collect)]
```

Comment:
That we have three `allow`s in our code base should be testament to the fact that this lint is a bit too false-positive-happy to be widely useful as is.

Luckily, it is certainly possible to improve it, so that shouldn't discourage you.